.github/workflows/ruby.yml

```diff
@@ -14,12 +14,18 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    # older versions of Ruby want openssl1 which is not on ubuntu-latest
+    runs-on: ubuntu-20.04
```

Comment:
Older versions of Ruby (incl 2.7) want openssl 1, downgrading the OS because I've had some slight hassles with installing packages using apt (I probably missed something obvious).